.github/workflows/build.yml

```diff
@@ -7,16 +7,48 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  generate_wheels_matrix:
+    name: Generate wheels matrix
+    runs-on: ubuntu-latest
+    outputs:
+      include: ${{ steps.set-matrix.outputs.include }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Checkout mypy
+        shell: bash
+        # use a commit hash checked into a file to get the mypy revision to build.
+        # submodules prove problematic since .git is outside cibuildwheel's manylinux container
+        run: |
+          git clone https://github.com/python/mypy.git --recurse-submodules
+          git -C mypy checkout $(cat mypy_commit)
+      - name: Install cibuildwheel
+        run: pipx install cibuildwheel==2.11.1
+      - id: set-matrix
+        run: |
+          MATRIX=$(
+            {
+              cibuildwheel --config-file=cibuildwheel.toml --print-build-identifiers --platform linux mypy \
+              | jq -Rc '{"only": inputs, "os": "ubuntu-latest"}' \
+              && cibuildwheel --config-file=cibuildwheel.toml --print-build-identifiers --platform macos mypy \
+              | jq -Rc '{"only": inputs, "os": "macos-latest"}' \
+              && cibuildwheel --config-file=cibuildwheel.toml --print-build-identifiers --platform windows mypy \
+              | jq -Rc '{"only": inputs, "os": "windows-latest"}'
+            } | jq -sc
+          )
+          echo "include=$MATRIX" | tee -a $GITHUB_OUTPUT
+        env:
+          CIBW_ARCHS_LINUX: x86_64
+          CIBW_ARCHS_MACOS: x86_64 arm64
+          CIBW_ARCHS_WINDOWS: AMD64
```

Comment:
macOS archs are also set in `cibuildwheel.toml`, maybe all archs should be set in the config file rather than as env vars?

Reply:
I think I answered this, but this list is what this CI supports, while `cibuildwheel.toml` is general - you can run it locally or on any other CI.